api/src/modules/scenarios/cost-template.service.ts

```diff
@@ -0,0 +1,16 @@
+import * as stream from 'stream';
+
+export const FileNotFound = Symbol();
+
+export const FileNotReady = Symbol();
+
+export const FilePiped = Symbol();
+
+export abstract class CostTemplateService {
+  abstract scheduleShapefileCostTemplateCreation(scenarioId: string): void;
+
+  abstract getShapefileCostTemplate(
+    scenarioId: string,
+    stream: stream.Writable,
+  ): Promise<typeof FileNotFound | typeof FileNotReady | typeof FilePiped>;
```

Comment:
the structure looks good - I would just suggest to tweak names a bit for extra clarity:

* `ScenarioCostSurfaceTemplateService`
* `scheduleCostSurfaceTemplateShapefileCreation()` (or given the class name as context, even just `scheduleTemplateShapefileCreation()`)
* `getCostSurfaceTemplateShapefile()` (or just `getTemplateShapefile()`)

Reply:
I wanted to avoid so-called smurf naming, but it depends what convention do you prefer.

I can rename it :)

Reply:
0bc3f766

Reply:
yep, dropping the useless smurf prefixes from member functions as in your commit is definitely best 😃